src/func_cache.c

```diff
@@ -91,6 +84,19 @@ time_bucket_sort_transform(FuncExpr *func)
 	return (Expr *) copyObject(second);
 }
 
+static Expr *
+time_bucket_sort_transform(FuncExpr *func)
+{
+	Assert(list_length(func->args) >= 2);
+	/*
+	 * If period and offset are not constants we must not do the optimization
+	 */
+	if (!time_bucket_has_const_period_and_offset(func))
```

Comment:
this has to be different for time_bucket and time_bucket_gapfill

Reply:
@svenklemm how so? `period` and `offset` look like they're in the same place, and from what I can tell the other arguments shouldn't matter for this optimization

Reply:
actually they are not only period is the same, 3rd and 4th are start/stop for gapfill

Reply:
@svenklemm this function is never called in the gapfill case, only for the regular timebucket (though I did need to move the period check to the gapfill case)